redbot/cogs/warnings/warnings.py

```diff
@@ -301,19 +342,39 @@ def __init__(self, bot: Red):
 
         await warning_points_add_check(self.config, ctx, user, current_point_count)
         try:
+            dm = await self.config.guild(ctx.guild).toggle_dm()
+            if dm:
+                em = discord.Embed(
+                    title=_("Warning from {user}").format(user=ctx.author),
+                    description=reason_type["description"],
+                )
+                em.add_field(name=_("Points"), value=str(reason_type["points"]))
+                await user.send(
+                    _("You have received a warning in {guild_name}.").format(
+                        guild_name=ctx.guild.name
+                    ),
+                    embed=em,
+                )
+        except discord.HTTPException:
+            await ctx.send(
+                "The warning couldn't be sent to the user because the user has DM's from server members turned off."
+            )
+        await ctx.send(_("User {user} has been warned in their DM's.").format(user=user))
```

Comment:
In my opinion, this message is unnecessary (as user that used this command should already know what settings he has), but if you want to keep it, put this in `else` block and check if DMs are enabled (cause otherwise, this message will be sent even if the warning couldn't be sent to the user or when sending DMs is disabled in cog's settings):
```suggestion
        else:
            if dm:
                await ctx.send(_("User {user} has been warned in their DMs.").format(user=user))
```